var/spack/repos/builtin/packages/proj/package.py

```diff
@@ -144,14 +148,18 @@ def configure_args(self):
             args.append("--with-external-gtest")
 
         if self.spec.satisfies("@7:"):
-            if "+tiff" in self.spec:
-                args.append("--enable-tiff")
-            else:
-                args.append("--disable-tiff")
+            args.extend(self.enable_or_disable("tiff"))
 
             if "+curl" in self.spec:
                 args.append("--with-curl=" + self.spec["curl"].prefix.bin.join("curl-config"))
             else:
                 args.append("--without-curl")
 
+        args.extend(self.enable_or_disable("shared"))
+        args.extend(self.with_or_without("pic"))
+
+        if self.spec.satisfies("^libtiff+jpeg~shared"):
```

Comment:
Can you comment more on why this is needed? I'm guessing this is a bug in the build system that will never be fixed since autotools is no longer supported?

Reply:
It's not exactly an autotools issue per se, it's that if I want to build with all-static libraries, then the lib/link flags from the 2nd and 3rd order dependencies don't automatically get incorporated. Unfortunately in my testing I'm running into some issues (probably a new dependency), so this will probably take a bit more work...

Reply:
You can't use `proj~~static`?

Reply:
I'm not sure what you mean? As an example, `spack install proj~shared ^libtiff~shared ^libjpeg-turbo~shared ^zlib~shared` does not work (with autotools or with cmake); it will fail with undefined references for jpeg and zlib functions. For autotools probably the easiest solution will be to use pkgconfig, but I'm not sure I can make it work with cmake unless I can magically tack on some flags to the end of each compiler call...

Reply:
Gotcha, sounds like a bug in the build system then. I'm fine with this hack, although it would be good to report to the devs if it also occurs for CMake.